xml_converter/src/xml_converter.cpp

```diff
@@ -62,40 +62,62 @@ vector<string> get_files_by_suffix(string directory, string suffix) {
     return files;
 }
 
-void read_taco_directory(
+map<string, vector<string>> read_taco_directory(
     string input_path,
     map<string, Category>* marker_categories,
     vector<Parseable*>* parsed_pois) {
+    map<string, vector<string>> category_to_filepath;
     if (!filesystem::exists(input_path)) {
         cout << "Error: " << input_path << " is not an existing directory or file" << endl;
     }
     else if (filesystem::is_directory(input_path)) {
+        string directory_name = filesystem::path(input_path).filename();
         vector<string> xml_files = get_files_by_suffix(input_path, ".xml");
         for (const string& path : xml_files) {
-            parse_xml_file(path, input_path, marker_categories, parsed_pois);
+            set<string> category_names = (parse_xml_file(path, input_path, marker_categories, parsed_pois));
+            string relative_path = directory_name + path.substr(input_path.size());
+            for (set<string>::iterator it = category_names.begin(); it != category_names.end(); it++) {
+                category_to_filepath[*it].push_back(relative_path);
+            }
         }
     }
     else if (filesystem::is_regular_file(input_path)) {
-        parse_xml_file(input_path, get_base_dir(input_path), marker_categories, parsed_pois);
+        set<string> category_names = parse_xml_file(input_path, get_base_dir(input_path), marker_categories, parsed_pois);
```

Comment:
```suggestion
        set<string> top_level_category_names = parse_xml_file(input_path, get_base_dir(input_path), marker_categories, parsed_pois);
```
same nit from the directory branch